### PR TITLE
[feg] add commit information in feg images. Add build.py to feg

### PR DESCRIPTION
**Pull request by uri200**
Signed-off-by: Oriol Batalla <obatalla@fb.com>

## Summary

Added some commit information to docker images in build time

A

Note this could have been done easier if DOCKERFILE would have some kind of conditonal copy. Thewn in case .git folder didnt exist, we could skip the creation of those envars. But this is not supported. That is why the need to create a pre build script

We took advantage of the need of that prebuild script and created a `build.py` similar to orc8r to build the images and run tests and precommit 

Commit information will show  doing `docker-compose exec magmad show_gateway_info.py`
```
root@docker-desktop:/# show_gateway_info.py

Hardware ID
-----------
f184da62-e944-4cf5-9805-6b18bae3f3d6

Challenge key
-------------
MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAEgls6FFUecrbtQTljawxCrR/JzDAwDmN4KwZbWSAk21pibCkEnNYSE1KlVkImoD02ULsPhZcw/9nwI7Z5UEOLfO2ueeQHT4gajSGDCfZ6O1F6HnAotd+p/s97+wEIRlDh

Version Info
-------------
Commit Branch: master
Commit Hash: 2cdcaa442
Commit Date: 2021-06-25 23:21:07 +0300

Notes
```

This will also show at start on all feg services
```
session_proxy    | ********
session_proxy    | Build Info
session_proxy    |  Commit Branch: master
session_proxy    |   Commit Tag:
session_proxy    |   Commit Hash: 2cdcaa442
session_proxy    |   Commit Date: 2021-06-25 23:21:07 +0300
session_proxy    | ********
```

## Test Plan

make precommit
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: feg/gateway/Makefile [hunk 1]
```diff
@@ -18,6 +20,7 @@ download:
 	go mod download
 
 install:
+	#go install -ldflags "-X main.MagmaGitCommit=$(MAGMA_GIT_COMMIT)"  ./... magma/gateway/services/magmad
```
Comment (uri200) [inline feg/gateway/Makefile:23]: will remove this soon

File: feg/gateway/services/session_proxy/session_proxy/main.go [hunk 2]
```diff
@@ -74,6 +79,8 @@ func main() {
 	lteprotos.RegisterCentralSessionControllerServer(srv.GrpcServer, sessionManagerAndHealthServer)
 	protos.RegisterServiceHealthServer(srv.GrpcServer, sessionManagerAndHealthServer)
 
+	glog.Fatal("DELMEEEE" )
```
Comment (uri200) [inline feg/gateway/services/session_proxy/session_proxy/main.go]: will remove this once done testing

File: feg/gateway/utils/configs.go [hunk 3]
```diff
@@ -117,3 +117,20 @@ func getFlagValue(flagName string) string {
 	}
 	return res
 }
+
+func PrintBuildInfo() string {
+	delimiter := "********"
```
Comment (emakeev) [inline feg/gateway/utils/configs.go:122]: const delimiter ...

File: orc8r/gateway/go/services/bootstrapper/gateway_info/get_info.go [hunk 4]
```diff
@@ -44,7 +45,22 @@ func GetFormatted() (string, error) {
 			config.GetMagmadConfigs().BootstrapConfig.ChallengeKey, err)
 		return res, err
 	}
-	res = fmt.Sprintf("%s\nChallenge Key:\n--------------\n%s\n",
-		res, base64.StdEncoding.EncodeToString(marshaledPubKey))
+	res = fmt.Sprintf("%s\nChallenge Key:\n--------------\n%s\n%s\n",
+		res, base64.StdEncoding.EncodeToString(marshaledPubKey), PrintBuildInfo())
```
Comment (emakeev) [inline orc8r/gateway/go/services/bootstrapper/gateway_info/get_info.go:49]: it's better to add delimiter() to magmad's main() - this is bootstrapping GW info function

File: feg/gateway/utils/configs.go [hunk 5]
```diff
@@ -117,3 +117,20 @@ func getFlagValue(flagName string) string {
...
+		"\tCommit Hash: %s\n"+
+		"\tCommit Date: %s\n"+
+		"%s\n",
+		delimiter,
+		os.Getenv("MAGMA_BUILD_BRANCH"),
+		os.Getenv("MAGMA_BUILD_TAG"),
+		os.Getenv("MAGMA_BUILD_COMMIT_HASH"),
+		os.Getenv("MAGMA_BUILD_COMMIT_DATE"),
+		delimiter)
+}
```
Comment (emakeev) [inline feg/gateway/utils/configs.go:136]: wouldn't it be easier & more reliable to embed all this information into Go binaries instead?

something like:

package main
...
var GitCommitHash string
...

and then buld with:

export GIT_COMMIT_HASH=$(git rev-list -1 HEAD) && go build -ldflags "-X main.GitCommitHash=$GIT_COMMIT_HASH" ...

**Review by rdefosse — APPROVED** [review]
I approved for the `.circleci/config.yml`

It would be nice that you pass the python linter check on this new python file you've added.

Raphael

**Comment by uri200** [comment]
splitting this PR. 

This should be the first one: https://github.com/magma/magma/pull/7836

**Comment by themarwhal** [comment]
Hey @uri200! Does this PR still need a review? (nvm noticed it's in draft mode, I'll ignore it for now :) )

**Comment by ekowtaylor** [comment]
@uri200, this PR has now gone stale. To maintain a healthy codebase, it will be automatically closed unless there are plans to rebase and merge by close of week

**Pull request closed without merging**